src/lib.rs

```diff
@@ -668,6 +668,73 @@ impl Hash for i24 {
     }
 }
 
+macro_rules! impl_i24_from_primitive {
+    ($($target_type:ty)*) => {
+	    $(
+		    impl core::convert::From<$target_type> for i24 {
+			    fn from(value: $target_type) -> Self {
+			        i24::from_i32(i32::from(value))
+			            .expect("From is only implemented for signed types with less than 24 bits")
```

Comment:
this leads to pretty bad code gen especially for bool

Reply:
It was meant to be a temporary solution for my use case; I'll optimize it later

Reply:
dw, fix coming soon been busy lately tho, so took a bit